In [1]:
import pandas as pd

In [2]:
%load_ext numpy_html

In [3]:
%%html
<style>
table {margin-top: 16pt;}
td {font-family: Arial, Helvetica, sans-serif !important; font-weight: 600; white-space: normal !important;}
tr:nth-child(even) {background: #b0d9eb !important; color: black}
tr:nth-child(odd) {background: #8fb1bf !important; color: black}
</style>

In [4]:
# Read the Excel file into a DataFrame
df = pd.read_csv("loans.csv")

# Display the first 5 rows
df.head()

Loan ID                           Customer ID  \
0  14dd8831-6af5-400b-83ec-68e61888a048  981165ec-3274-42f5-a3b4-d104041a9ca9   
1  4771cc26-131a-45db-b5aa-537ea4ba5342  2de017a3-2e01-49cb-a581-08169e83be29   
2  4eed4e6a-aa2f-4c91-8651-ce984ee8fb26  5efb2b2b-bf11-4dfd-a572-3761a2694725   
3  77598f7b-32e7-4e3b-a6e5-06ba0d98fe8a  e777faab-98ae-45af-9a86-7ce5b33b1011   
4  d4062e70-befa-4995-8643-a0de73938182  81536ad9-5ccf-4eb8-befb-47a4d608658e   

  Loan Status  Current Loan Amount        Term  Credit Score  Annual Income  \
0  Fully Paid             445412.0  Short Term         709.0      1167493.0   
1  Fully Paid             262328.0  Short Term           NaN            NaN   
2  Fully Paid           99999999.0  Short Term         741.0      2231892.0   
3  Fully Paid             347666.0   Long Term         721.0       806949.0   
4  Fully Paid             176220.0  Short Term           NaN            NaN   

  Years in current job Home Ownership             Purpose  Monthly Debt  \
0              8 years  Home Mortgage   Home Improvements       5214.74   
1            10+ years  Home Mortgage  Debt Consolidation      33295.98   
2              8 years       Own Home  Debt Consolidation      29200.53   
3              3 years       Own Home  Debt Consolidation       8741.90   
4              5 years           Rent  Debt Consolidation      20639.70   

   Years of Credit History  Months since last delinquent  \
0                     17.2                           NaN   
1                     21.1                           8.0   
2                     14.9                          29.0   
3                     12.0                           NaN   
4                      6.1                           NaN   

   Number of Open Accounts  Number of Credit Problems  Current Credit Balance  \
0                      6.0                        1.0                228190.0   
1                     35.0                        0.0                229976.0   
2                     18.0                        1.0                297996.0   
3                      9.0                        0.0                256329.0   
4                     15.0                        0.0                253460.0   

   Maximum Open Credit  Bankruptcies  Tax Liens  
0             416746.0           1.0        0.0  
1             850784.0           0.0        0.0  
2             750090.0           0.0        0.0  
3             386958.0           0.0        0.0  
4             427174.0           0.0        0.0

In [5]:
max = 99999999
df = df[df['Current Loan Amount'] < max]

In [6]:
# df = df.dropna(subset=['Annual Income'])
average = df['Current Loan Amount'].mean()
df['Annual Income'] = df['Annual Income'].fillna(average)

In [7]:
min1 = df['Annual Income'].min()
min1

76627.0

In [8]:
max1 = df['Annual Income'].max()
max1

165557393.0

In [9]:
ID1 = 'bbf87a87-22cd-4d10-bd9b-7a9cc1b6e59d'
result1 = df.loc[df['Loan ID'] == ID1, 'Home Ownership'].values[0]
result1

'Own Home'

In [10]:
df['Actual Annual Income'] = df['Annual Income'] - 12 * df['Monthly Debt']

In [11]:
ID2 = '76fa89b9-e6a8-49af-afa1-8151315aba8e'
result2 = df.loc[df['Loan ID'] == ID2, 'Actual Annual Income'].values[0]
result2

674234.76

In [12]:
inc = df.loc[df['Actual Annual Income'].idxmin()]
id = inc['Loan ID']
id

'0a84b8ab-ed2f-474f-a3a9-8bf5bb90c31a'

In [13]:
long_term = (df['Term'] == 'Long Term').sum()
long_term

25475

In [14]:
bankr = (df['Bankruptcies'] > 1).sum()
bankr

476

In [15]:
df = df[df['Purpose'] == 'Home Improvements']
short_term = (df['Term'] == 'Short Term').sum()
short_term

3490

In [16]:
uniq = df['Purpose'].nunique()
uniq

1

In [17]:
mc = df['Purpose'].describe()
mc

count                  5124
unique                    1
top       Home Improvements
freq                   5124
Name: Purpose, dtype: object

In [18]:
most_common_values = df['Purpose'].value_counts().nlargest(3)
most_common_values

Purpose
Home Improvements    5124
Name: count, dtype: int64

In [23]:
#correlation_matrix = df.corr(numeric_only=True)
#correlation_matrix

cor1 = df['Annual Income'].corr(df['Number of Open Accounts'])
cor1

0.11768251999868755

In [25]:
cor2 = df['Number of Credit Problems'].corr(df['Bankruptcies'])
cor2

0.7796246291033961

<b>There is a high correlation between 'Number of Credit Problems' and 'Bankruptcies', (0.78) which is obvious
since if you have problems with credits, you will likely to go bankrupt soon. 'Annual Income' and 
'Number of Open Accounts' do not have almost any correlation (0.12), I guess it does not really matter
how much you make, you still need a bank account in your life.</b>